Userland/Libraries/LibJS/Print.h

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2020-2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2020-2022, Linus Groh <linusg@serenityos.org>
+ * Copyright (c) 2022, Ali Mohammad Pur <mpfard@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/HashTable.h>
+#include <LibCore/Stream.h>
+#include <LibJS/Runtime/Value.h>
+
+namespace JS {
+struct PrintContext {
+    JS::VM& vm;
+    Core::Stream::Stream& stream;
+    bool strip_ansi { false };
+};
+
+ErrorOr<void> print(JS::Value value, PrintContext);
```

Comment:
This signature looks kinda funny. Isn't that print context non copyable and non moveable? How can we pass by value?

Reply:
Why would it be non-copyable?

Reply:
Reference data members? Doesn't having them delete all the special members?

Reply:
I'll make it take the context by ref, but I'm still not sure why you expect it to be non-copyable, it's just a bunch of references and a bool - the copy is just memcpy()

Reply:
> Doesn't having them delete all the special members?

only rvalue refs for copy/move ctors, "any" reference type only deletes the default ctor afaik

Reply:
see "Deleted implicitly-declared copy constructor" in  https://en.cppreference.com/w/cpp/language/copy_constructor.